### PR TITLE
Form work for EAD object associations

### PR DESCRIPTION
**Pull request by mghughes**

### CODE REVIEW

File: includes/utilities.inc [hunk 1]
```diff
@@ -22,6 +22,34 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+/**
+ * Autocomplete callback; return all objects that can be an association.
+ */
+function islandora_ead_get_potential_associations($search_value) {
+  $tuque = islandora_get_tuque_connection();
+  $query = <<<EOQ
```
Comment (jordandukart) [inline includes/utilities.inc]: Any concern with the size of the RI and the time it'll take to perform this query as opposed to using Solr?

File: includes/utilities.inc [hunk 2]
```diff
@@ -48,3 +76,66 @@ EOQ;
...
+ *   The PID of the EAD object to retrieve the association of.
+ * @param string $refid
+ *   The level ID to retrieve the association of.
+ *
+ * @return string
+ *   The PID of the associated object or NULL.
+ */
+function islandora_ead_retrieve_refid_association($eadid, $refid) {
+  $tuque = islandora_get_tuque_connection();
+  $sparql_query = <<<EOQ
```
Comment (jordandukart) [inline includes/utilities.inc]: Same as https://github.com/discoverygarden/islandora_solution_pack_ead/pull/2/files#r255143103

File: includes/utilities.inc [hunk 3]
```diff
@@ -48,3 +76,66 @@ EOQ;
...
+
+  $ead_dom = new DOMDocument();
+  $ead_dom->loadXML($ead['EAD']->content);
+  $ead_dom = $xslt->transformToDoc($ead_dom);
+
+  $xpath = new DOMXPath($ead_dom);
+  $entity = $xpath->query("/entities/entity[id = '$refid']")->item(0);
+  $containers = array();
+  foreach ($xpath->query('container', $entity) as $container) {
+    $type = $xpath->query('type', $container)->item(0)->textContent;
```
Comment (jordandukart) [inline includes/utilities.inc]: Could do this with `->evaluate('string(whatever)', $container)`. Also any concern with these values being empty?

File: includes/association_existing_object.form.inc [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    '#autocomplete_path' => "islandora/object/$ead->id/manage/ead/get_potential_associations",
+    '#default_value' => $association,
+    '#required' => TRUE,
+    '#type' => 'textfield',
+    '#title' => t('Select an object'),
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Link Object'),
+  );
```
Comment (jordandukart) [inline includes/association_existing_object.form.inc:58]: Is a removal option (and optionally purging the object) missing to this form?

File: includes/association_existing_object.form.inc [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ *   The drupal form definition.
+ */
+function islandora_ead_association_existing_object_form(array $form, array &$form_state, AbstractObject $ead, $refid) {
+  module_load_include('inc', 'islandora_ead', 'includes/utilities');
+  $form_state['ead'] = $ead;
+  $form_state['refid'] = $refid;
+
+  $level_info = islandora_ead_get_level_info($ead, $refid);
+  $form['description'] = array(
+    '#markup' => t('Select the object to link to !level. Leave blank to remove current association.', array(
```
Comment (jordandukart) [inline includes/association_existing_object.form.inc:30]: Did we run this UI past @desmodude750? Not sure if we have a standard way of doing something like this. Seems like could just add a remove button that triggers a separate submit handler which could be clearer?

File: includes/utilities.inc [hunk 6]
```diff
@@ -22,6 +22,38 @@
...
+      '@label' => $label,
+      '@pid' => $data['solr_doc']['PID'],
+    ));
+  };
+
+  $results = array();
+  foreach ($qp->islandoraSolrResult['response']['objects'] as $result) {
+    $results[$result['PID']] = $construct_label($result);
+  }
+  ksort($results);
```
Comment (jordandukart) [inline includes/utilities.inc]: Not sure this `ksort` is necessary here? Could take advantage of Solr's sort field such that we don't need to sort in memory in PHP.

File: includes/association_existing_object.form.inc (hunk 5 above)
Comment (mghughes) [reply includes/association_existing_object.form.inc:30]: @desmodude750 unsure about it as well, but @mohirt seemed to think this approach is better than the two buttons. And this current submit handler will delete the old association when you provide a new one anyway, so the behaviour seems consistent when you don't provide a new one.

**Pull request merged**